lib/constructs/callback-interface.js

```diff
@@ -178,14 +197,41 @@ class CallbackInterface {
   }
 
   generateInstall() {
+    if (this.constants.size > 0) {
+      this.str += `
+        const exposed = new Set([
+      `;
+
+      for (const globalName of this.exposed) {
+        this.str += `"${globalName}",\n`;
+      }
+
+      this.str += `
+        ]);
+      `;
+    }
+
     this.str += `
-      exports.install = function install(globalObject) {
+      exports.install = (globalObject, globalNames) => {
     `;
 
     if (this.constants.size > 0) {
       const { name } = this;
 
       this.str += `
+        let isExposed = false;
+
+        for (const globalName of globalNames) {
+          if (exposed.has(globalName)) {
+            isExposed = true;
+            break;
+          }
+        }
+
+        if (!isExposed) {
+          return;
+        }
```

Comment:
Can we simplify it to just `if (!globalNames.some(name => exposed.has(name)) return;`

Reply:
`globalNames` is a `Set` in <https://github.com/jsdom/webidl2js/pull/190>.

Reply:
Ah, someone should really make a `Set.prototype.some`/`every` proposal to TC39. But I'd rather the API be taking an array.

Reply:
There is the **Iterator Helpers** proposal: <https://tc39.es/proposal-iterator-helpers/>.